From 6773ee0387f34b3b26ca5c0be6d410305ef533d2 Mon Sep 17 00:00:00 2001
From: Philip Withnall <withnall@endlessm.com>
Date: Wed, 13 Mar 2019 16:08:32 +0000
Subject: [PATCH] build: Stop building libglib-testing as a subproject
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Just add it as a dependency instead. It’s a lot less painful (git
submodules are still a pain to use; and `git evtag` doesn’t work well
with them); and libglib-testing has just done a 0.1.0 release which we
can depend on.

Signed-off-by: Philip Withnall <withnall@endlessm.com>
---
 .gitlab-ci.yml                  | 1 -
 .gitmodules                     | 3 ---
 libmalcontent/tests/meson.build | 2 +-
 meson.build                     | 6 ------
 subprojects/libglib-testing     | 1 -
 5 files changed, 1 insertion(+), 12 deletions(-)
 delete mode 100644 .gitmodules
 delete mode 160000 subprojects/libglib-testing

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -21,7 +21,6 @@ fedora:
   except:
     - tags
   script:
-    - git submodule update --init
     - meson --buildtype debug --werror -Db_coverage=true -Ddocumentation=true _build .
     - meson test -C _build
     # FIXME: lcov doesn't support gcc9 yet:
--- a/.gitmodules
+++ /dev/null
@@ -1,3 +0,0 @@
-[submodule "subprojects/libglib-testing"]
-	path = subprojects/libglib-testing
-	url = https://gitlab.gnome.org/pwithnall/libglib-testing.git
--- a/libmalcontent/tests/meson.build
+++ b/libmalcontent/tests/meson.build
@@ -3,8 +3,8 @@ deps = [
   dependency('gio-unix-2.0', version: '>= 2.44'),
   dependency('glib-2.0', version: '>= 2.60.0'),
   dependency('gobject-2.0', version: '>= 2.44'),
+  dependency('glib-testing-0'),
   libmalcontent_dep,
-  libglib_testing_dep,
 ]
 
 envs = test_env + [
--- a/meson.build
+++ b/meson.build
@@ -40,12 +40,6 @@ polkit_gobject = dependency('polkit-gobj
 polkitpolicydir = polkit_gobject.get_pkgconfig_variable('policydir',
   define_variable: ['prefix', prefix])
 
-libglib_testing = subproject('libglib-testing')
-libglib_testing_dep = dependency(
-  'libglib-testing',
-  fallback: ['libglib-testing', 'libglib_testing_dep'],
-)
-
 config_h = configuration_data()
 config_h.set_quoted('GETTEXT_PACKAGE', 'malcontent')
 config_h.set_quoted('PACKAGE_LOCALE_DIR', join_paths(get_option('prefix'), get_option('localedir')))
